pkg/hostagent/pod_relatives.go

```diff
@@ -181,6 +183,90 @@ func (agent *HostAgent) initNetPolPodIndex() {
 	})
 }
 
+func (agent *HostAgent) initQoSPolicyInformerFromClient(
+	qosClient *qospolicyclset.Clientset) {
+	agent.initQoSPolicyInformerBase(
+		cache.NewListWatchFromClient(
+			qosClient.AciV1().RESTClient(), "qospolicies",
+			metav1.NamespaceAll, fields.Everything()))
+}
+
+func (agent *HostAgent) initQoSPolicyInformerBase(listWatch *cache.ListWatch) {
+	agent.qosPolicyInformer =
+		cache.NewSharedIndexInformer(
+			listWatch, &qospolicy.QosPolicy{}, 0,
+			cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		)
+
+	agent.qosPolicyInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			agent.qosPolicyAdded(obj)
+		},
+		UpdateFunc: func(oldobj interface{}, newobj interface{}) {
+			agent.qosPolicyChanged(oldobj, newobj)
+		},
+		DeleteFunc: func(obj interface{}) {
+			agent.qosPolicyDeleted(obj)
+		},
+	},
+	)
+}
+
+func (agent *HostAgent) qosPolicyAdded(obj interface{}) {
+	agent.qosPolPods.UpdateSelectorObj(obj)
+}
+
+func (agent *HostAgent) qosPolicyChanged(oldobj, newobj interface{}) {
+	oldnp := oldobj.(*qospolicy.QosPolicy)
+	newnp := newobj.(*qospolicy.QosPolicy)
+	if !reflect.DeepEqual(oldnp.Spec.Selector, newnp.Spec.Selector) {
+		agent.qosPolPods.UpdateSelectorObjNoCallback(newobj)
+	}
+
+	npkey, err := cache.MetaNamespaceKeyFunc(newnp)
```

Comment:
NIt:  can you change this name to npkey to qpkey

Reply:
done